http_log_writer.go

```diff
@@ -191,48 +200,68 @@ func (w *httpLogWriter) flush() {
 			panic("non-empty buffer shouldn't return an error on Read")
 		}
 
-		part := httpLogPart{
+		logPartsPayload = append(logPartsPayload, &httpLogPart{
 			JobID:   w.jobID,
 			Content: string(buf[0:n]),
 			Number:  w.logPartNumber,
-		}
+		})
 		w.logPartNumber++
 
-		err = w.publishLogPart(part)
-		if err != nil {
-			context.LoggerFromContext(w.ctx).WithFields(logrus.Fields{
-				"err":  err,
-				"self": "http_log_writer",
-			}).Error("couldn't publish log part")
+		if len(logPartsPayload) >= w.maxBatchSize {
+			w.logFlushError(w.publishLogParts(logPartsPayload))
+			logPartsPayload = []*httpLogPart{}
 		}
 	}
-}
 
-func (w *httpLogWriter) publishLogPart(part httpLogPart) error {
-	part.UUID, _ = context.UUIDFromContext(w.ctx)
+	w.logFlushError(w.publishLogParts(logPartsPayload))
+}
 
-	type logPartPayload struct {
-		Type     string `json:"@type"`
-		Final    bool   `json:"final"`
-		Content  string `json:"content"`
-		Encoding string `json:"encoding"`
+func (w *httpLogWriter) logFlushError(err error) {
+	if err == nil {
+		return
 	}
 
-	payload := logPartPayload{
+	context.LoggerFromContext(w.ctx).WithFields(logrus.Fields{
+		"err":  err,
+		"self": "http_log_writer",
+	}).Error("couldn't publish log parts")
```

Comment:
Looks like we're already covered with the logrus sentry hook if `sentry-hook-errors` is enabled. 👍